nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -352,33 +391,98 @@ public void nestedVariableDeclarationChecks() {
             "import org.jspecify.annotations.Nullable;",
             "class Test {",
             "  class D<P extends @Nullable Object> {}",
-            "  class B<P extends @Nullable Object> extends D<P>{}",
-            "  class C<P extends @Nullable Object>{}",
-            "  class A<T extends C<P>, P extends @Nullable Object>{}",
-            " void sampleError() {",
-            "   // BUG: Diagnostic contains: Cannot assign from type",
-            "   D<C<String>> f1 = new B<C<@Nullable String>>();",
-            "   // BUG: Diagnostic contains: Cannot assign from type",
-            "   A<C<String>, String> f2 = new A<C<String>, @Nullable String>();",
-            "   // BUG: Diagnostic contains: Cannot assign from type",
-            "   D<C<String>> f3 = new B<@Nullable C<String>>();",
-            " }",
-            "  }")
+            "  class B<P extends @Nullable Object> extends D<P> {}",
+            "  class C<P extends @Nullable Object> {}",
+            "  class A<T extends C<P>, P extends @Nullable Object> {}",
+            "  void testPositive() {",
+            "    // BUG: Diagnostic contains: Cannot assign from type",
+            "    D<C<String>> f1 = new B<C<@Nullable String>>();",
+            "    // BUG: Diagnostic contains: Cannot assign from type",
+            "    A<C<String>, String> f2 = new A<C<String>, @Nullable String>();",
+            "    // BUG: Diagnostic contains: Cannot assign from type",
+            "    D<C<String>> f3 = new B<@Nullable C<String>>();",
+            "  }",
+            "  void testNegative() {",
+            "    D<C<@Nullable String>> f1 = new B<C<@Nullable String>>();",
+            "    A<C<String>, String> f2 = new A<C<String>, String>();",
+            "    D<@Nullable C<String>> f3 = new B<@Nullable C<String>>();",
+            "  }",
+            "}")
         .doTest();
   }
 
-  /** check that we don't report errors on invalid instantiations in unannotated code */
   @Test
-  public void instantiationInUnannotatedCode() {
+  public void testForMethodReferenceInAnAssignment() {
     makeHelper()
         .addSourceLines(
             "Test.java",
-            "package com.other;",
+            "package com.uber;",
             "import org.jspecify.annotations.Nullable;",
             "class Test {",
-            "  static class NonNullTypeParam<E> { }",
-            "  static void instOf(Object o) {",
-            "    Object p = (NonNullTypeParam<@Nullable String>) o;",
+            "  interface A<T1 extends @Nullable Object> {",
+            "    String function(T1 o);",
+            "  }",
+            "  static String foo(Object o) {",
+            "    return o.toString();",
+            "  }",
+            "  static void testPositive() {",
+            "    // we should report an error here, since Test::foo cannot take",
+            "    // a @Nullable parameter.  we don't catch this yet",
```

Comment:
👍 Good enough for the moment, but glad we have the reminder/todo here. Actually, maybe we can start this comment with an actual "// TODO: we should report [...]"? 🙂